### DISCUSSION TITLE
Using velero and minio, when using the example with pv, it gives an error

### DISCUSSION THREAD

**Discussion by dylanyht** [vmware-tanzu/velero#3512] [Community support Q&A]
**What steps did you take and what happened:**
velero v1.5.3

velero install  --provider aws --plugins velero/velero-plugin-for-aws:v1.1.0 --bucket  velero \
--secret-file ./credentials-velero \
--use-restic \
--default-volumes-to-restic \
--backup-location-config \
region=minio,s3ForcePathStyle="true",s3Url=http://172.16.77.64:9000

I used with-pv.yaml for backup recovery.
After I executed velero backup create nginx-backup --include-namespaces nginx-example, I checked the logs and found that an error was reported

The logs are as follows

# velero backup logs nginx-backup |grep error
time="2021-03-01T01:42:06Z" level=error msg="Error executing hook" backup=velero/nginx-backup error="command terminated with exit code 1" hookPhase=pre hookSource=annotation hookType=exec logSource="internal/hook/item_hook_handler.go:206" name=nginx-deployment-5ccc99bffb-q54nr namespace=nginx-example resource=pods
time="2021-03-01T01:42:06Z" level=error msg="Error backing up item" backup=velero/nginx-backup error="command terminated with exit code 1" logSource="pkg/backup/backup.go:437" name=nginx-deployment-5ccc99bffb-q54nr
time="2021-03-01T01:42:06Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/nginx-backup error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/github.com/vmware-tanzu/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:449" name=pvc-0faa5e10-0ac3-43fd-acf6-db44a8c8a995 namespace= persistentVolume=pvc-0faa5e10-0ac3-43fd-acf6-db44a8c8a995 resource=persistentvolumes volumeSnapshotLocation=default

cnetos7
kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:43:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by carlisia** [comment]
Please tell me more about your environment. Is this trying to create an AWS snapshot? Did you intent on using restic for that?

**Reply by dylanyht** [reply]
I'm installing velero with Restic enabled, using minio storage, not AWS. I'm using https://github.com/vmware-tanzu/velero/blob/main/examples/nginx-app/with-pv.yaml for this example Error when creating a backup, this example contains a PVC

**Comment by carlisia** [comment]
I noticed this error in your log:

```
 msg="Error getting volume snapshotter for volume snapshot location" backup=velero/nginx-backup error="rpc error: code = Unknown desc = missing region in aws configuration"
```

Have you checked your VSL configuration? Here's an example: https://velero.io/docs/v1.5/api-types/volumesnapshotlocation/.

**Reply by dylanyht** [reply]
I checked the VSL configuration and there is indeed no AWS zone configuration in it. But the minio I built with private cloud does not use AWS. so I can ignore this problem? Here is the complete VSL configuration

```
apiVersion: velero.io/v1
kind: VolumeSnapshotLocation
metadata:
  labels:
    component: velero
  name: default
  namespace: velero
spec:
  provider: aws

```

**Reply by dsu-igeek** [reply]
The AWS plug-in is used for any S3 compatible storage.  There's a blog post here that describes the entire setup: https://velero.io/blog/velero-v1-1-on-vsphere/

If you're still having issues, please provide complete logs.  Some of the info we need to debug the problem will be in the non-error messages.

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`


**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Velero version (use `velero version`):
- Velero features (use `velero client config get features`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

Restic Daemonset logs

`kubectl logs -n velero daemonset.apps/restic`

**Comment by r10r** [comment]
Hey @dylanyht, I did run into the same issue when trying out velero with CSI support. 

```
time="2021-08-10T12:50:55Z" level=error msg="Error executing hook" backup=velero/nginx-backup1 error="command terminated with exit code 1" hookPhase=pre hookSource=annotation hookType=exec logSource="internal/hook/item_hook_handler.go:206" name=nginx-deployment-66689547d-k7rw5 namespace=nginx-example resource=pods
```

I solved this by removing the following fsfreeze related annotations  from the `nginx-example` pod.
After that the backup completes successfully.

```
post.hook.backup.velero.io/command: ["/sbin/fsfreeze", "--unfreeze", "/var/log/nginx"]
post.hook.backup.velero.io/container: fsfreeze
pre.hook.backup.velero.io/command: ["/sbin/fsfreeze", "--freeze", "/var/log/nginx"]
pre.hook.backup.velero.io/container: fsfreeze
```

My VolumeSnapshotClass is:

```
apiVersion: snapshot.storage.k8s.io/v1
kind: VolumeSnapshotClass
metadata:
  name: netapp-snapclass
  annotations:
    snapshot.storage.kubernetes.io/is-default-class: "true"
  labels:
    velero.io/csi-volumesnapshot-class: "true"
driver: csi.trident.netapp.io
deletionPolicy: Retain
parameters:
```

`fsfreeze` only supports the following filesystems (see `man 8 fsfreeze`)

> fsfreeze - suspend access to an filesystem (Linux Ext3/4, ReiserFS, JFS, XFS).

So fsfreeze will likely fail if the snapshot class driver uses a different filesystem  to mount volumes (e.g  NFS for `csi.trident.netapp.io`)

**Reply by frouzbeh** [reply]
Hi @r10r, I'm trying to install velero with aws and csi plugin. In the overriding values.yaml I set `snapshotEnabled = false` to prevent velero from creating the volumesnapshot crds and instead using the csi snapshotclass. But I see backup gets partially failed with this error:
`time="2021-10-11T16:40:24Z" level=error msg="Error getting volume snapshotter for volume snapshot location" backup=velero/velero-daily-monitoring-20211011164017 error="rpc error: code = Unknown desc = missing region in aws configuration" error.file="/go/src/velero-plugin-for-aws/velero-plugin-for-aws/volume_snapshotter.go:76" error.function="main.(*VolumeSnapshotter).Init" logSource="pkg/backup/item_backupper.go:449" name=pvc-81d5709f-b985-4c9a-b601-d277d6d57561 namespace= persistentVolume=pvc-81d5709f-b985-4c9a-b601-d277d6d57561 resource=persistentvolumes volumeSnapshotLocation=default`

Would you mind sharing your values.yaml? Thanks

**Comment by r10r** [comment]
Hi @frouzbeh,  I did run into this issue too. You might try the following:

* From [Can't execute persistentvolumesnapshot with csi](https://github.com/vmware-tanzu/velero/issues/3836)
* See also https://velero.io/docs/v1.6/api-types/volumesnapshotlocation/

```sh
kubectl delete volumesnapshotlocation.velero.io -n velero default
velero snapshot-location create default --provider aws --config region=minio
```
